Cannot use schemaRegistry with multiple URL

**Describe the bug**
From the [documentation](https://www.jikkou.io/docs/providers/schema-registry/configuration/#configuration), we should be able to specify a list of schema registry url separated by a comma:
```
# Comma-separated list of URLs for schema registry instances that can be used to register or look up schemas
```
But I'm not able to have it working (but maybe I'm doing something wrong ?).

**To Reproduce**
Create a Jikkou config with multiple urls for schema registry, like:
```
jikkou {
  schemaRegistry {
    url = "http://host-a:8081,http://host-b:8081"
  }
}
```
Then try to use it
```
$ jikkou get schemaregistrysubjects
io.streamthoughts.jikkou.core.exceptions.JikkouRuntimeException: Failed to list all schema registry subject versions
        at io.streamthoughts.jikkou.schema.registry.reconciler.SchemaRegistrySubjectCollector.listAll(SchemaRegistrySubjectCollector.java:114)
        at io.streamthoughts.jikkou.schema.registry.reconciler.SchemaRegistrySubjectCollector.listAll(SchemaRegistrySubjectCollector.java:84)
        at io.streamthoughts.jikkou.core.DefaultApi.listResources(DefaultApi.java:569)
        at io.streamthoughts.jikkou.client.command.get.GetResourceCommand.call(GetResourceCommand.java:74)
        at io.streamthoughts.jikkou.client.command.get.GetResourceCommand.call(GetResourceCommand.java:29)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
        at io.streamthoughts.jikkou.client.Jikkou.executionStrategy(Jikkou.java:150)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at io.streamthoughts.jikkou.client.Jikkou.execute(Jikkou.java:140)
        at io.streamthoughts.jikkou.client.Jikkou.main(Jikkou.java:128)
        at java.base@21.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: io.streamthoughts.jikkou.http.client.RestClientException: Error at index 4 in: "8081,http:"
        at io.streamthoughts.jikkou.http.client.internal.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:122)
        at jdk.proxy4/jdk.proxy4.$Proxy63.listSubjects(Unknown Source)
        at reactor.core.publisher.MonoCallable.call(MonoCallable.java:72)
        at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:128)
        at reactor.core.publisher.MonoFlatMapMany.subscribeOrReturn(MonoFlatMapMany.java:49)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4552)
        at reactor.core.publisher.Mono.block(Mono.java:1778)
        at io.streamthoughts.jikkou.schema.registry.reconciler.SchemaRegistrySubjectCollector.listAll(SchemaRegistrySubjectCollector.java:112)
        ... 16 more
        Suppressed: java.lang.Exception: #block terminated with an error
                at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:104)
                at reactor.core.publisher.Mono.block(Mono.java:1779)
                ... 17 more
Caused by: java.net.MalformedURLException: Error at index 4 in: "8081,http:"
        at java.base@21.0.2/java.net.URL.<init>(URL.java:806)
        at java.base@21.0.2/java.net.URL.of(URL.java:913)
        at java.base@21.0.2/java.net.URI.toURL(URI.java:1172)
        at org.glassfish.jersey.client.internal.HttpUrlConnector._apply(HttpUrlConnector.java:371)
        at org.glassfish.jersey.client.internal.HttpUrlConnector.apply(HttpUrlConnector.java:273)
        at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:300)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$2(JerseyInvocation.java:699)
        at org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:709)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:703)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:391)
        at org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:703)
        at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:698)
        at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:432)
        at org.glassfish.jersey.client.proxy.WebResourceFactory.invoke(WebResourceFactory.java:272)
        at io.streamthoughts.jikkou.http.client.internal.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:110)
        ... 23 more
Caused by: java.lang.NumberFormatException: Error at index 4 in: "8081,http:"
        at java.base@21.0.2/java.lang.Integer.parseInt(Integer.java:743)
        at java.base@21.0.2/java.net.URLStreamHandler.parseURL(URLStreamHandler.java:236)
        at java.base@21.0.2/java.net.URL.<init>(URL.java:801)
        ... 40 more

Error: JikkouRuntimeException: Failed to list all schema registry subject versions
```

**Expected behavior**
Jikkou should be able to handle multiple URL for schema registry.

**Runtime environment**
 - Jikkou: 0.35.7

--- a/providers/jikkou-provider-schema-registry/src/integration-test/java/io/streamthoughts/jikkou/schema/registry/health/SchemaRegistryHealthIndicatorTest.java
+++ b/providers/jikkou-provider-schema-registry/src/integration-test/java/io/streamthoughts/jikkou/schema/registry/health/SchemaRegistryHealthIndicatorTest.java
@@ -10,6 +10,7 @@ import io.streamthoughts.jikkou.core.health.Health;
 import io.streamthoughts.jikkou.core.models.ApiHealthResult;
 import io.streamthoughts.jikkou.schema.registry.BaseExtensionProviderIT;
 import java.time.Duration;
+import java.util.List;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
@@ -23,7 +24,7 @@ class SchemaRegistryHealthIndicatorTest extends BaseExtensionProviderIT {
         Health expected = Health.builder()
             .up()
             .name("schemaregistry")
-            .details("schema.registry.url", schemaRegistryUrl())
+            .details("schema.registry.url", List.of(schemaRegistryUrl()))
             .details("http.response.status", 200)
             .build();
         Assertions.assertEquals(ApiHealthResult.from(expected), result);

--- a/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/SchemaRegistryExtensionProvider.java
+++ b/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/SchemaRegistryExtensionProvider.java
@@ -29,6 +29,8 @@ import io.streamthoughts.jikkou.schema.registry.validation.CompatibilityLevelVal
 import io.streamthoughts.jikkou.schema.registry.validation.SchemaCompatibilityValidation;
 import io.streamthoughts.jikkou.schema.registry.validation.SubjectNameRegexValidation;
 import io.streamthoughts.jikkou.spi.BaseExtensionProvider;
+import java.util.Arrays;
+import java.util.List;
 import org.jetbrains.annotations.NotNull;
 
 /**
@@ -83,8 +85,12 @@ public final class SchemaRegistryExtensionProvider extends BaseExtensionProvider
     @Override
     public void configure(@NotNull Configuration configuration) throws ConfigException {
         super.configure(configuration);
+        List<String> urls = Arrays.stream(Config.SCHEMA_REGISTRY_URL.get(configuration).split(","))
+            .map(String::trim)
+            .filter(s -> !s.isEmpty())
+            .toList();
         this.clientConfig = new SchemaRegistryClientConfig(
-            Config.SCHEMA_REGISTRY_URL.get(configuration),
+            urls,
             Config.SCHEMA_REGISTRY_VENDOR.get(configuration),
             Config.SCHEMA_REGISTRY_AUTH_METHOD.get(configuration),
             () -> Config.SCHEMA_REGISTRY_BASIC_AUTH_USER.get(configuration),

--- a/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/api/FailoverSchemaRegistryApi.java
+++ b/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/api/FailoverSchemaRegistryApi.java
@@ -1,0 +1,198 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) The original authors
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.streamthoughts.jikkou.schema.registry.api;
+
+import io.streamthoughts.jikkou.schema.registry.api.data.CompatibilityCheck;
+import io.streamthoughts.jikkou.schema.registry.api.data.CompatibilityLevelObject;
+import io.streamthoughts.jikkou.schema.registry.api.data.CompatibilityObject;
+import io.streamthoughts.jikkou.schema.registry.api.data.ModeObject;
+import io.streamthoughts.jikkou.schema.registry.api.data.SchemaString;
+import io.streamthoughts.jikkou.schema.registry.api.data.SubjectSchemaId;
+import io.streamthoughts.jikkou.schema.registry.api.data.SubjectSchemaRegistration;
+import io.streamthoughts.jikkou.schema.registry.api.data.SubjectSchemaVersion;
+import io.streamthoughts.jikkou.schema.registry.api.data.SubjectVersion;
+import jakarta.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link SchemaRegistryApi} decorator that provides failover across multiple Schema Registry instances.
+ * On connection-related failures, the next instance in the list is tried.
+ */
+public final class FailoverSchemaRegistryApi implements SchemaRegistryApi {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FailoverSchemaRegistryApi.class);
+
+    private final List<SchemaRegistryApi> delegates;
+
+    /**
+     * Creates a new {@link FailoverSchemaRegistryApi} instance.
+     *
+     * @param delegates the list of Schema Registry API instances to failover between.
+     */
+    public FailoverSchemaRegistryApi(List<SchemaRegistryApi> delegates) {
+        if (delegates == null || delegates.isEmpty()) {
+            throw new IllegalArgumentException("At least one SchemaRegistryApi instance is required");
+        }
+        this.delegates = List.copyOf(delegates);
+    }
+
+    private <T> T executeWithFailover(Function<SchemaRegistryApi, T> action) {
+        RuntimeException lastException = null;
+        for (int i = 0; i < delegates.size(); i++) {
+            SchemaRegistryApi delegate = delegates.get(i);
+            try {
+                return action.apply(delegate);
+            } catch (RuntimeException e) {
+                if (isConnectionFailure(e)) {
+                    LOG.warn("Connection failed to Schema Registry instance [{}], trying next instance", i, e);
+                    lastException = e;
+                } else {
+                    throw e;
+                }
+            }
+        }
+        throw lastException;
+    }
+
+    /**
+     * Determines if an exception represents a connection failure
+     * by checking the cause chain for {@link IOException}.
+     */
+    private static boolean isConnectionFailure(RuntimeException e) {
+        Throwable cause = e;
+        while (cause != null) {
+            if (cause instanceof IOException) {
+                return true;
+            }
+            cause = cause.getCause();
+        }
+        return false;
+    }
+
+    @Override
+    public Response get() {
+        return executeWithFailover(SchemaRegistryApi::get);
+    }
+
+    @Override
+    public List<String> listSubjects() {
+        return executeWithFailover(SchemaRegistryApi::listSubjects);
+    }
+
+    @Override
+    public List<Integer> deleteSubjectVersions(String subject, boolean permanent) {
+        return executeWithFailover(api -> api.deleteSubjectVersions(subject, permanent));
+    }
+
+    @Override
+    public List<Integer> getAllSubjectVersions(String subject) {
+        return executeWithFailover(api -> api.getAllSubjectVersions(subject));
+    }
+
+    @Override
+    public SubjectSchemaId registerSchema(String subject, SubjectSchemaRegistration schema, boolean normalize) {
+        return executeWithFailover(api -> api.registerSchema(subject, schema, normalize));
+    }
+
+    @Override
+    public SubjectSchemaVersion checkSubjectVersion(String subject, SubjectSchemaRegistration schema, boolean normalize) {
+        return executeWithFailover(api -> api.checkSubjectVersion(subject, schema, normalize));
+    }
+
+    @Override
+    public SubjectSchemaVersion getLatestSubjectSchema(String subject) {
+        return executeWithFailover(api -> api.getLatestSubjectSchema(subject));
+    }
+
+    @Override
+    public SubjectSchemaVersion getSchemaByVersion(String subject, int version) {
+        return executeWithFailover(api -> api.getSchemaByVersion(subject, version));
+    }
+
+    @Override
+    public List<String> getSchemasTypes() {
+        return executeWithFailover(SchemaRegistryApi::getSchemasTypes);
+    }
+
+    @Override
+    public SchemaString getSchemaById(String id) {
+        return executeWithFailover(api -> api.getSchemaById(id));
+    }
+
+    @Override
+    public String getSchemaOnlyById(String id) {
+        return executeWithFailover(api -> api.getSchemaOnlyById(id));
+    }
+
+    @Override
+    public List<SubjectVersion> getVersionSchemaById(String id) {
+        return executeWithFailover(api -> api.getVersionSchemaById(id));
+    }
+
+    @Override
+    public CompatibilityLevelObject getGlobalCompatibility() {
+        return executeWithFailover(SchemaRegistryApi::getGlobalCompatibility);
+    }
+
+    @Override
+    public CompatibilityLevelObject getConfigCompatibility(String subject, boolean defaultToGlobal) {
+        return executeWithFailover(api -> api.getConfigCompatibility(subject, defaultToGlobal));
+    }
+
+    @Override
+    public CompatibilityObject updateConfigCompatibility(String subject, CompatibilityObject compatibility) {
+        return executeWithFailover(api -> api.updateConfigCompatibility(subject, compatibility));
+    }
+
+    @Override
+    public CompatibilityObject deleteConfigCompatibility(String subject) {
+        return executeWithFailover(api -> api.deleteConfigCompatibility(subject));
+    }
+
+    @Override
+    public ModeObject getMode() {
+        return executeWithFailover(SchemaRegistryApi::getMode);
+    }
+
+    @Override
+    public ModeObject getMode(String subject) {
+        return executeWithFailover(api -> api.getMode(subject));
+    }
+
+    @Override
+    public ModeObject updateMode(String subject, ModeObject mode) {
+        return executeWithFailover(api -> api.updateMode(subject, mode));
+    }
+
+    @Override
+    public ModeObject deleteMode(String subject) {
+        return executeWithFailover(api -> api.deleteMode(subject));
+    }
+
+    @Override
+    public CompatibilityCheck testCompatibility(String subject, int version, boolean verbose,
+                                                 SubjectSchemaRegistration schema) {
+        return executeWithFailover(api -> api.testCompatibility(subject, version, verbose, schema));
+    }
+
+    @Override
+    public CompatibilityCheck testCompatibilityLatest(String subject, boolean verbose,
+                                                       SubjectSchemaRegistration schema) {
+        return executeWithFailover(api -> api.testCompatibilityLatest(subject, verbose, schema));
+    }
+
+    @Override
+    public void close() {
+        for (SchemaRegistryApi delegate : delegates) {
+            delegate.close();
+        }
+    }
+}

--- a/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/api/SchemaRegistryApiFactory.java
+++ b/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/api/SchemaRegistryApiFactory.java
@@ -10,6 +10,7 @@ import io.streamthoughts.jikkou.common.utils.Encoding;
 import io.streamthoughts.jikkou.http.client.RestClientBuilder;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
+import java.util.List;
 import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -23,12 +24,24 @@ public final class SchemaRegistryApiFactory {
 
     /**
      * Creates a new {@link SchemaRegistryApi} for the given configuration.
+     * If multiple URLs are configured, a {@link FailoverSchemaRegistryApi} is returned
+     * that tries each URL in order on connection failure.
      *
      * @param config the configuration.
      * @return a new {@link SchemaRegistryApi} instance.
      */
     public static SchemaRegistryApi create(SchemaRegistryClientConfig config) {
-        URI baseUri = URI.create(config.url());
+        List<String> urls = config.urls();
+        if (urls.size() == 1) {
+            return createForUrl(URI.create(urls.getFirst()), config);
+        }
+        List<SchemaRegistryApi> delegates = urls.stream()
+                .map(url -> createForUrl(URI.create(url), config))
+                .toList();
+        return new FailoverSchemaRegistryApi(delegates);
+    }
+
+    private static SchemaRegistryApi createForUrl(URI baseUri, SchemaRegistryClientConfig config) {
         LOG.info("Create new Schema Registry client for: {}", baseUri);
         RestClientBuilder builder = RestClientBuilder
                 .newBuilder()

--- a/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/api/SchemaRegistryClientConfig.java
+++ b/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/api/SchemaRegistryClientConfig.java
@@ -7,10 +7,11 @@
 package io.streamthoughts.jikkou.schema.registry.api;
 
 import io.streamthoughts.jikkou.http.client.ssl.SSLConfig;
+import java.util.List;
 import java.util.function.Supplier;
 
 public record SchemaRegistryClientConfig(
-    String url,
+    List<String> urls,
     String vendor,
     AuthMethod authMethod,
     Supplier<String> basicAuthUser,
@@ -19,4 +20,12 @@ public record SchemaRegistryClientConfig(
     Boolean debugLoggingEnabled
 ) {
 
+    /**
+     * Returns the first URL from the list.
+     *
+     * @return the first configured URL.
+     */
+    public String firstUrl() {
+        return urls.getFirst();
+    }
 }

--- a/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/health/SchemaRegistryHealthIndicator.java
+++ b/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/health/SchemaRegistryHealthIndicator.java
@@ -85,7 +85,7 @@ public final class SchemaRegistryHealthIndicator implements HealthIndicator {
                 builder = builder.details("http.response.status", response.getStatus());
             }
             return builder
-                    .details("schema.registry.url", config.url())
+                    .details("schema.registry.url", config.urls())
                     .build();
         } finally {
             api.close();

--- a/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/reconciler/SchemaRegistrySubjectCollector.java
+++ b/providers/jikkou-provider-schema-registry/src/main/java/io/streamthoughts/jikkou/schema/registry/reconciler/SchemaRegistrySubjectCollector.java
@@ -81,7 +81,7 @@ public class SchemaRegistrySubjectCollector extends ContextualExtension implemen
         this.config = config;
         this.schemaRegistrySubjectFactory = new V1SchemaRegistrySubjectFactory(
             config.vendor(),
-            config.url(),
+            config.firstUrl(),
             prettyPrintSchema
         );
     }

--- a/providers/jikkou-provider-schema-registry/src/test/java/io/streamthoughts/jikkou/schema/registry/api/SchemaRegistryApiFactoryTest.java
+++ b/providers/jikkou-provider-schema-registry/src/test/java/io/streamthoughts/jikkou/schema/registry/api/SchemaRegistryApiFactoryTest.java
@@ -42,7 +42,7 @@ class SchemaRegistryApiFactoryTest {
                 .build();
         schemaRegistryMockServer.setDispatcher(schemaRegistryHTTPDispatcher);
         SchemaRegistryClientConfig config = new SchemaRegistryClientConfig(
-                String.format("http://%s:%s", schemaRegistryMockServer.getHostName(), schemaRegistryMockServer.getPort()),
+                List.of(String.format("http://%s:%s", schemaRegistryMockServer.getHostName(), schemaRegistryMockServer.getPort())),
                 "generic",
                 AuthMethod.BASICAUTH,
                 () -> "username",
@@ -62,6 +62,94 @@ class SchemaRegistryApiFactoryTest {
         String authorization = schemaRegistryMockServer.takeRequest().getHeaders().get("Authorization");
         // result should correspond to base64 encoded string "username:password" prefixed with "Basic"
         Assertions.assertEquals(authorization, "Basic dXNlcm5hbWU6cGFzc3dvcmQ=");
+    }
+
+    @Test
+    @DisplayName("Should failover to second URL when first server is down")
+    public void shouldFailoverToSecondUrl() throws IOException {
+        // Given
+        MockWebServer downServer = new MockWebServer();
+        downServer.start();
+        int downPort = downServer.getPort();
+        downServer.close(); // shut it down to simulate unreachable server
+
+        MockWebServer upServer = new MockWebServer();
+        upServer.start();
+        HttpPathBasedDispatcher dispatcher = HttpPathBasedDispatcher.builder()
+                .forPath("/subjects", new MockResponse.Builder()
+                        .code(200)
+                        .addHeader("Content-Type", "application/vnd.schemaregistry.v1+json")
+                        .body("[\"subject-1\",\"subject-2\"]")
+                        .build())
+                .build();
+        upServer.setDispatcher(dispatcher);
+
+        SchemaRegistryClientConfig config = new SchemaRegistryClientConfig(
+                List.of(
+                        String.format("http://localhost:%s", downPort),
+                        String.format("http://%s:%s", upServer.getHostName(), upServer.getPort())
+                ),
+                "generic",
+                AuthMethod.NONE,
+                () -> null,
+                () -> null,
+                () -> SSLConfig.from(Configuration.empty()),
+                false
+        );
+
+        // When
+        List<String> subjects;
+        try (SchemaRegistryApi schemaRegistryApi = SchemaRegistryApiFactory.create(config)) {
+            subjects = schemaRegistryApi.listSubjects();
+        }
+
+        // Then
+        Assertions.assertNotNull(subjects);
+        Assertions.assertEquals(List.of("subject-1", "subject-2"), subjects);
+
+        upServer.close();
+    }
+
+    @Test
+    @DisplayName("Should create single client when only one URL is configured")
+    public void shouldCreateSingleClientForOneUrl() {
+        // Given
+        SchemaRegistryClientConfig config = new SchemaRegistryClientConfig(
+                List.of(String.format("http://%s:%s", schemaRegistryMockServer.getHostName(), schemaRegistryMockServer.getPort())),
+                "generic",
+                AuthMethod.NONE,
+                () -> null,
+                () -> null,
+                () -> SSLConfig.from(Configuration.empty()),
+                false
+        );
+
+        // When
+        try (SchemaRegistryApi api = SchemaRegistryApiFactory.create(config)) {
+            // Then - should not be a FailoverSchemaRegistryApi
+            Assertions.assertFalse(api instanceof FailoverSchemaRegistryApi);
+        }
+    }
+
+    @Test
+    @DisplayName("Should create failover client when multiple URLs are configured")
+    public void shouldCreateFailoverClientForMultipleUrls() {
+        // Given
+        SchemaRegistryClientConfig config = new SchemaRegistryClientConfig(
+                List.of("http://host-a:8081", "http://host-b:8081"),
+                "generic",
+                AuthMethod.NONE,
+                () -> null,
+                () -> null,
+                () -> SSLConfig.from(Configuration.empty()),
+                false
+        );
+
+        // When
+        try (SchemaRegistryApi api = SchemaRegistryApiFactory.create(config)) {
+            // Then - should be a FailoverSchemaRegistryApi
+            Assertions.assertInstanceOf(FailoverSchemaRegistryApi.class, api);
+        }
     }
 
     @AfterAll

